src/main/java/seedu/address/logic/commands/SearchCommand.java

```diff
@@ -0,0 +1,229 @@
+         */
+        public void setTags(Set<Tag> tags) {
+            this.tags = (tags != null) ? new HashSet<>(tags) : null;
+        }
+
+        /**
+         * Returns an unmodifiable tag set, which throws {@code UnsupportedOperationException}
+         * if modification is attempted.
+         * Returns {@code Optional#empty()} if {@code tags} is null.
+         */
+        public Optional<Set<Tag>> getTags() {
+            return (tags != null) ? Optional.of(Collections.unmodifiableSet(tags)) : Optional.empty();
+        }
+
+        public SearchPredicate getPredicate() {
+            ArrayList<ContainsKeywordsPredicate> predicateList = new ArrayList<>();
+            ContainsKeywordsPredicate<Name> nameSearch =
+                    new ContainsKeywordsPredicate<>(PREFIX_NAME, this.getName());
+            predicateList.add(nameSearch);
+            ContainsKeywordsPredicate<Phone> phoneSearch =
+                    new ContainsKeywordsPredicate<>(PREFIX_PHONE, this.getPhone());
+            predicateList.add(phoneSearch);
+            ContainsKeywordsPredicate<Email> emailSearch =
+                    new ContainsKeywordsPredicate<>(PREFIX_EMAIL, this.getEmail());
+            predicateList.add(emailSearch);
+            ContainsKeywordsPredicate<Country> countrySearch =
+                    new ContainsKeywordsPredicate<>(PREFIX_COUNTRY, this.getCountry());
+            predicateList.add(countrySearch);
+            ContainsKeywordsPredicate<Status> statusSearch =
+                    new ContainsKeywordsPredicate<>(PREFIX_STATUS, this.getStatus());
+            predicateList.add(statusSearch);
+            ContainsKeywordsPredicate<Comment> commentSearch =
+                    new ContainsKeywordsPredicate<>(PREFIX_COMMENT, this.getComment());
+            predicateList.add(commentSearch);
+            ContainsKeywordsPredicate<Set<Tag>> tagSearch =
+                    new ContainsKeywordsPredicate<>(PREFIX_TAG, this.getTags());
+            predicateList.add(tagSearch);
+            return new SearchPredicate(predicateList);
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            if (other == this) {
+                return true;
+            }
+
+            // instanceof handles nulls
+            if (!(other instanceof SearchCommand.SearchPersonDescriptor)) {
+                return false;
+            }
+
+            SearchCommand.SearchPersonDescriptor otherSearchPersonDescriptor =
+                    (SearchCommand.SearchPersonDescriptor) other;
+            return Objects.equals(name, otherSearchPersonDescriptor.name)
+                    && Objects.equals(phone, otherSearchPersonDescriptor.phone)
+                    && Objects.equals(email, otherSearchPersonDescriptor.email)
+                    && Objects.equals(country, otherSearchPersonDescriptor.country)
+                    && Objects.equals(comment, otherSearchPersonDescriptor.comment)
```

Comment:
forgot to add Objects.equals(status, otherSearchPersonDescriptor.status)??

Reply:
Thanks for the comment, will fix soon